### PR TITLE
Fix unnecessary home directory creation issues

### PR DESCRIPTION
**Pull request by roy-dydx**
### Changelist
- Revert https://github.com/dydxprotocol/v4-chain/pull/1062 temporary fix and bring in official cosmos fix for it
- Pass in a temp dir in client tempApp to prevent creation of `data` directory in the cwd

### Test Plan
- Tested query commands to ensure correct home directory was being used in default and non-default cases. Checked to make sure default one wasn't created when not intended.
- Ran local net and inspected containers to make sure data directory isn't created in `/dydxprotocol/`.

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: protocol/cmd/dydxprotocold/cmd/root.go [hunk 1]
```diff
@@ -90,7 +91,7 @@ func NewRootCmdWithInterceptors(
 		dbm.NewMemDB(),
 		nil,
 		true,
-		viper.New(),
```
Comment (lcwik) [inline protocol/cmd/dydxprotocold/cmd/root.go:93]: Can this be replaced with:
```
initAppOptions := viper.New()
initAppOptions.Set(flags.FlagHome, tempDir())
tempApp(..., initAppOptions)
```

Was hoping to keep away from test utilities in Cosmos SDK as part of the production code.

**Review by lcwik — APPROVED** [review]
There is an existing ticket about get command line tests for coverage of these edge cases since it is difficult to know what the initial client context and config should be since the Cosmos SDK documentation is lacking as to how those values are needed for Auto CLI opts. Is the additional tests you can add in a follow-up?

**Comment by lcwik** [comment]
There is a lint issue with an import of the same statement twice and tests are failing (one seems to be due to a flake)

**Comment by roy-dydx** [comment]
@Mergifyio backport release/protocol/v4.x
